app/models/concerns/course/assessment/submission/todo_concern.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+module Course::Assessment::Submission::TodoConcern
+  extend ActiveSupport::Concern
+
+  included do
+    after_create :update_todo
+    after_destroy :restart_todo
+
+    define_method :finalise do |*args|
+      super(*args) if defined?(super)
+      return unless todo.in_progress?
+      todo.complete!
+      todo.save!
+    end
+
+    define_method :unsubmit do |*args|
```

Comment:
It's ok, think it's better to document this part, otherwise it's a bit implicit.